osquery/config/config.cpp

```diff
@@ -64,6 +68,36 @@ DECLARE_string(config_plugin);
 DECLARE_string(pack_delimiter);
 DECLARE_bool(disable_events);
 
+std::string hashBuffer(HashType hash_type, const void* buffer, size_t size) {
```

Comment:
Ignoring for the moment the messy state of the rest of the touched files, here's a POC outlining my current thought process. I know this isn't the most elegant solution, but it seems much preferable to the idea of extending the hash table to allow it to hash contents from memory. Any arguments to the contrary or alternative suggestions would be welcome.